### PR TITLE
optimize: use docker progress

### PR DESCRIPTION
**Pull request by justadogistaken**
### Describe what this PR does / why we need it
the original progress came across many problems, we decide to abandon it, and use docker progress to support progress output.

### Special notes for reviews
After this pr merged, layer hash store in image yaml will be like shaxxx:fff2826b39e284bdae16853d3922c868b1aea0b41cdd0e1c005c2be9db001d37.
But layer dir name is still fff2826b39e284bdae16853d3922c868b1aea0b41cdd0e1c005c2be9db001d37 only.

### CODE REVIEW

File: image/util.go [hunk 1]
```diff
@@ -110,3 +111,22 @@ func GetClusterFileFromBaseImage(imageName string) string {
 	}
 	return string(data)
 }
+
+func getDockerAuthInfoFromDocker(domain string) types.AuthConfig {
```
Comment (allencloud) [inline image/util.go:115]: I do not think that making username/password empty when failure happens is reasonable. It is not so friendly for users.

File: image/store/layer.go [hunk 2]
```diff
@@ -13,29 +13,34 @@ type LayerID digest.Digest
...
 	return string(li)
 }
+
+func NewROLayer(LayerDigest digest.Digest, size int64) (Layer, error) {
+	err := LayerDigest.Validate()
+	if err != nil {
+		return nil, err
+	}
+	return &roLayer{id: LayerID(LayerDigest), size: size}, nil
+}
```
Comment (allencloud) [inline image/store/layer.go:46]: wrap line please

File: build/local_builder.go [hunk 3]
```diff
@@ -284,6 +289,21 @@ func (l *LocalBuilder) UpdateImageMetadata() error {
 		l.Image.Annotations[common.ImageAnnotationForClusterfile] = string(bytes)
 	}
 
+	// save layer ids
+	for _, layer := range l.Image.Spec.Layers {
+		if layer.Hash != "" {
```
Comment (kakaZhou719) [inline build/local_builder.go:294]: LGTM

**Review by kakaZhou719 — APPROVED** [review]

File: image/distributionutil/pull.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+			LayerErr = layerStore.RegisterLayerIfNotPresent(roLayer)
+			if LayerErr != nil {
+				errorCh <- LayerErr
+				return
+			}
+		}(l)
+	}
+	done.Wait()
+	if len(errorCh) > 0 {
```
Comment (touchuyht) [inline image/distributionutil/pull.go:73]: Maybe it's better to put close(errorCh) outside the if statement.

**Pull request merged**